docker/docker_image_dest.go

```diff
@@ -120,15 +151,25 @@ func (d *dockerImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobI
 			return types.BlobInfo{Digest: inputInfo.Digest, Size: size}, nil
 		}
 	}
+	mountQuery := prepareMountQuery(d.mounts, inputInfo.Digest.String())
 
 	// FIXME? Chunked upload, progress reporting, etc.
 	uploadPath := fmt.Sprintf(blobUploadPath, reference.Path(d.ref.ref))
+	if mountQuery != "" {
+		logrus.Debugf("Trying mounts: %s", mountQuery)
+		uploadPath = fmt.Sprintf("%s?%s", uploadPath, mountQuery)
+	}
 	logrus.Debugf("Uploading %s", uploadPath)
 	res, err := d.c.makeRequest(context.TODO(), "POST", uploadPath, nil, nil)
 	if err != nil {
 		return types.BlobInfo{}, err
 	}
 	defer res.Body.Close()
+	if res.StatusCode == http.StatusCreated {
+		logrus.Debugf("Layer mounted: %s", inputInfo.Digest)
+		return types.BlobInfo{Digest: inputInfo.Digest}, nil
```

Comment:
`PutBlob`, if successful, must also set the `Size` member of `types.BlobInfo` (so that schema2 manifests can be created).

Reply:
Fixed.